manticore/__main__.py

```diff
@@ -93,20 +111,40 @@ def positive(value):
 
 
 def ethereum_cli(args):
-    from .ethereum import ManticoreEVM, IntegerOverflow, UninitializedStorage, UninitializedMemory
+    from .ethereum import ManticoreEVM, DetectInvalid, DetectIntegerOverflow, DetectUninitializedStorage, DetectUninitializedMemory, FilterFunctions
     log.init_logging()
 
     m = ManticoreEVM(procs=args.procs)
 
-    ################ Default? Detectors #######################
-    m.register_detector(IntegerOverflow())
-    m.register_detector(UninitializedStorage())
-    m.register_detector(UninitializedMemory())
+    if args.detect_all:
+        args.detect_invalid = True
+        args.detect_io = True
+        args.detect_uninitialized_storage = True
+        args.detect_uninitialized_memory = True
+
+    if args.detect_invalid:
+        m.register_detector(DetectInvalid())
+    if args.detect_io:
+        m.register_detector(DetectIntegerOverflow())
+    if args.detect_uninitialized_storage:
+        m.register_detector(DetectUninitializedStorage())
+    if args.detect_uninitialized_memory:
+        m.register_detector(DetectUninitializedMemory())
+
+    if args.avoid_constant:
+        #avoid all human level tx that has no effect on the storage
+        print "avoiding contants"
+        filter_nohuman_constants = FilterFunctions(regexp=r".*", depth='human', mutability='constant', include=False)
+        self.register_plugin(filter_nohuman_constants)
 
     logger.info("Beginning analysis")
 
     m.multi_tx_analysis(args.argv[0], args.contract, args.txlimit, not args.txnocoverage, args.txaccount)
 
+    #TODO unregister all plugins
+
+    m.finalize()
```

Comment:
If there needs to be some 'destruction part' maybe we should implement `__enter__` and `__exit__` and use `with statement` e.g. `with Manticore(...) as m:`

Reply:
It may be the case you do not want to call that .finalize().
Maybe the naming is off. That will go through all the states in the queues and generate a report for each and remove each from the queues. 
Sometimes you want to do that browsing by hand and report only on some states (as opposed to all of them).

But maybe... we could have the enter/exit AND also have something to clean the lists so at __exit__ it does not do anything on those special cases. Need to think about it.